roxie/ccd/ccdserver.cpp

```diff
@@ -5207,16 +5226,20 @@ class CRemoteResultAdaptor : implements IEngineRowStream, implements IFinalRoxie
                     }
                 }
             }
-            else
+            else if (!anyActivity && !localAgent && !acknowledgeAllRequests)
             {
                 unsigned timeNow = msTick();
-                if (!anyActivity && !localAgent && (timeNow-lastActivity >= timeout))
+                if (timeNow-lastActivity >= checkInterval)
                 {
                     lastActivity = timeNow;
-                    activity.queryLogCtx().CTXLOG("Input has stalled for %u ms - retry required?", timeout);
-                    retryPending();
+                    activity.queryLogCtx().CTXLOG("Input has stalled for %u ms - retry required?", checkInterval);
+                    retryPending(0);
                 }
             }
+            if (acknowledgeAllRequests && !localAgent)
```

Comment:
Could this if block go before the if(mr) (at line 4945) ?

Reply:
It could, but I'm not sure it should. Would mean for example that you would do a resend before discovering that the message you just received was the ack you were waiting for...

Reply:
ok, good point.  I was concerned with the case where the if (mr) block returns and then retryPending() is not checked/called.
But I dont know how much of an issue that really is or if we could just also check retryPending() before the return.
Or just leave it and merge as is.

Reply:
Perhaps I should move it to the top of the loop then, before calling getNextResult()